GitCommands/IDiffListSortService.cs

```diff
@@ -0,0 +1,57 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reactive.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace GitCommands
+{
+    public enum DiffListSortType
+    {
+        /// <summary>
+        /// Invalid default
+        /// </summary>
+        Invalid,
+
+        /// <summary>
+        /// Sorts by file path alphanumerically
+        /// </summary>
+        FilePath,
+
+        /// <summary>
+        /// Sorts by file extension then by path
+        /// </summary>
+        FileExtension,
+
+        /// <summary>
+        /// Sorts by git change type. Addition, Deletions, edits, etc. then by path
+        /// </summary>
+        FileStatus
+    }
+
+    public interface IDiffListSortService
+    {
+        DiffListSortType DiffListSorting { get; set; }
+
+        event EventHandler DiffListSortingChanged;
+    }
+
+    public static class DiffListSortServiceExtensions
+    {
+        /// <summary>
+        /// Provides the <see cref="IDiffListSortService.DiffListSorting"/> immediately and then an element for each <see cref="IDiffListSortService.DiffListSortingChanged"/>.
+        /// </summary>
+        /// <param name="diffListSortService">The diff list service</param>
+        /// <returns>A hot stream with one immediate cold element.</returns>
+        public static IObservable<DiffListSortType> CurrentAndFutureSorting(this IDiffListSortService diffListSortService)
```

Comment:
Is there a reason this can't be a member of `IDiffListSortService`?

Reply:
Mostly because it doesn't have to be. Should be the same for any possible IDiffListSortService.